app/src/main/java/pl/droidevs/books/login/LoginViewModel.java

```diff
@@ -0,0 +1,37 @@
+package pl.droidevs.books.login;
+
+import android.arch.lifecycle.ViewModel;
+import android.content.SharedPreferences;
+
+import javax.inject.Inject;
+
+import pl.droidevs.books.validators.LoginValidator;
+
+public class LoginViewModel extends ViewModel {
+
+    private static final String SHARED_PREFERENCES_LOGIN_KEY = "login";
+
+    @Inject
+    SharedPreferences sharedPreferences;
+
+    @Inject
```

Comment:
W dokumentacji Daggera wyczytałem, że albo annotujemy pola, albo konstrutor:
`
class Thermosiphon implements Pump {
  private final Heater heater;

  @Inject
  Thermosiphon(Heater heater) {
    this.heater = heater;
  }
  ...
}
`

`class CoffeeMaker {
  @Inject Heater heater;
  @Inject Pump pump;
  ...
}`